lib/segment/src/segment_constructor/segment_constructor_base.rs

```diff
@@ -108,7 +108,8 @@ fn create_segment(
             )?,
         };
 
-        if config.quantization_config(vector_name).is_some() {
+        if config.quantization_config(vector_name).is_some() && config.quantization_config.is_some()
```

Comment:
I don't see how this change would prevent quantization on appendable segments. Could you elaborate?

Note that the `config.quantization_config(vector_name)` function returns the vector specific config, and if it doesn't exist, the global config.

Reply:
So when we were designing configuration for individual named vectors, we had in mind the following scenario:

configuration of individual vectors( if exists ) should be preferred over global configuration (if global configuration enabled)

See how indexing works: it only applies named configuration if there is a global one. But for quantization we broke this logic and tried to always load quantized data no matter if global config is set or not.

This new condition fixes his logic: we only load quantization if global config is set.

Worth mentioning that currently, if there quantization is configured in collection - it is always set in named config. 
So currently - each segment always tries to load quntization.

Reply:
Thanks for the explanation.

Then, shouldn't we change `config.quantization_config(vector_name)` to only return a configuration when a global one is set as well? Then we could keep this line intact, and possibly prevent confusion in the future.

Reply:
maybe, but I want to release the fix Asap, to fix the problem for users. refactoring can be done in minor release

Reply:
Okay! :smiley: